sklearn/manifold/tests/test_t_sne.py

```diff
@@ -286,11 +288,48 @@ def test_preserve_trustworthiness_approximately_with_precomputed_distances():
                     early_exaggeration=2.0, metric="precomputed",
                     random_state=i, verbose=0)
         X_embedded = tsne.fit_transform(D)
-        t = trustworthiness(D, X_embedded, n_neighbors=1,
-                            precomputed=True)
+        t = trustworthiness(D, X_embedded, n_neighbors=1, metric="precomputed")
         assert t > .95
 
 
+def test_trustworthiness_precomputed_deprecation():
+    # NOTE: Remove this test in v0.23
+
+    # Use of the flag `precomputed` in trustworthiness parameters has been
+    # deprecated, but will still work until v0.23.
+    X = np.arange(100).reshape(50, 2)
+    assert_equal(assert_warns(DeprecationWarning, trustworthiness,
+                              pairwise_distances(X), X, precomputed=True), 1.)
+    assert_equal(assert_warns(DeprecationWarning, trustworthiness,
+                              pairwise_distances(X), X, metric='precomputed',
+                              precomputed=True), 1.)
+    # If a 'metric' different from 'precomputed' is specified, but the flag
+    # 'precomputed' is set to True, the flag overwrites the parameter 'metric'
+    # and so the 'precomputed' metric will be used. Indeed, in
+    # http://scikit-learn.org/stable/developers/contributing.html#deprecation,
+    # the old parameter overwrites the new parameter.
+    assert_raises(Exception, assert_warns, DeprecationWarning,
+                  trustworthiness, X, X, metric='euclidean', precomputed=True)
+    assert_equal(assert_warns(DeprecationWarning, trustworthiness,
+                              pairwise_distances(X), X, metric='euclidean',
+                              precomputed=True), 1.)
+
+
+def test_trustworthiness_unusual_metric():
+    # Other metrics than 'euclidean' and 'precomputed' are unusual and must
+    # raise a warning
+    X = np.arange(100).reshape(50, 2)
+    assert_warns(RuntimeWarning, trustworthiness, X, X, metric='manhattan')
```

Comment:
I actually strongly believe we should not raise a warning here or make any statement that this is abnormal. Even if deriving the manifold from an unusual metric is weird, checking that neighborhoods are maintained in the embedded space is precisely how we determine whether the learnt manifold was appropriate.